common/djangoapps/student/views.py

```diff
@@ -855,6 +863,10 @@ def dashboard(request):
     valid_verification_statuses = ['approved', 'must_reverify', 'pending', 'expired']
     display_sidebar_on_dashboard = len(order_history_list) or verification_status in valid_verification_statuses
 
+    # Filter out any course enrollment course cards that are associated with fulfilled entitlements
+    for entitlement in [e for e in course_entitlements if e.enrollment_course_run is not None]:
+        course_enrollments = [enr for enr in course_enrollments if entitlement.enrollment_course_run.course_id != enr.course_id]  # pylint: disable=line-too-long
```

Comment:
Nit: this is a little difficult to read. Could it be simplified?

Reply:
Made it concise because this render function is already so long. Considered using:
>>> filter(lambda entitlement: entitlement.enrollment_course_run is not None, course_entitlements)
but that seemed even more complicated to understand